### PR TITLE
Connector Service API upgrade

### PR DESCRIPTION
**Pull request by Szymongib**
**Description**

Changes proposed in this pull request:

- Align Connector Service to new api

**Related issue(s)**
#2304

### CODE REVIEW

File: components/connector-service/internal/tokens/manager_test.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+		tokenGenerator := func() (string, apperrors.AppError) { return "", apperrors.Internal("error") }
+		tokenService := NewTokenManager(tokenTTL, nil, tokenGenerator)
+
+		_, err := tokenService.Save(serializable)
+
+		require.Error(t, err)
+	})
+}
+
+func TestTokenService_Replace(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (Szymongib) [reply components/connector-service/internal/tokens/manager_test.go]: Do we need that test, as it only calls the method on `store`?

File: components/connector-service/internal/clientcontext/extractor.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+func ExtractClusterContext(ctx context.Context) (ConnectorClientContext, apperrors.AppError) {
+	clusterCtx, ok := ctx.Value(ClusterContextKey).(ClusterContext)
+	if !ok {
+		return nil, apperrors.Internal("Failed to create params when reading ClusterContext")
+	}
+
+	return clusterCtx, nil
+}
+
+func ResolveClusterContextExtender(token string, tokenResolver tokens.Resolver) (ContextExtender, apperrors.AppError) {
```
[the remark this answers is not in the document]
Comment (akgalwas) [reply components/connector-service/internal/clientcontext/extractor.go]: Done

File: components/connector-service/internal/clientcontext/extractor.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	var clusterContext ClusterContext
+	err := tokenResolver.Resolve(token, &clusterContext)
+
+	return clusterContext, err
+}
+
+func ResolveApplicationContextExtender(token string, tokenResolver tokens.Resolver) (ContextExtender, apperrors.AppError) {
```
[the remark this answers is not in the document]
Comment (akgalwas) [reply components/connector-service/internal/clientcontext/extractor.go]: Done

File: components/connector-service/internal/tokens/manager.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+		tokenTTL:  tokenTTL,
+		store:     store,
+		generator: generator,
+	}
+}
+
+func (svc *tokenManager) Save(serializableContext Serializer) (string, apperrors.AppError) {
+	jsonData, err := serializableContext.ToJSON()
+	if err != nil {
+		return "", apperrors.Internal("Faild to serilize token params to JSON, %s", err.Error())
```
[the remark this answers is not in the document]
Comment (akgalwas) [reply components/connector-service/internal/tokens/manager.go]: Done

File: components/connector-service/internal/clientcontext/model.go [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+	return appCtx.Application
+}
+
+// GetCommonName returns expected Common Name value for the Application
+func (appCtx ApplicationContext) GetCommonName() string {
+	// TODO - adjust CN format after decision is made
+	return appCtx.Application
+}
+
+type ClusterContext struct {
```
[the remark this answers is not in the document]
Comment (akgalwas) [reply components/connector-service/internal/clientcontext/model.go]: Agreed to stay with the current terminology.

File: components/connector-service/README.md [hunk 6]
```diff
@@ -25,9 +25,16 @@ The Connector Service has the following parameters:
 - **internalAPIPort** - This port exposes the Connector Service within Kubernetes cluster. The default port is `8080`.
 - **namespace** - Namespace where Connector Service is deployed. The default namespace is `kyma-integration`.
 - **tokenLength** - Length of registration tokens. The default value is `64`.
-- **tokenExpirationMinutes** - Time after which tokens expire and are no longer valid. The default value is `5` minutes.
-- **domainName** - Domain name of the cluster, used for URL generating. Default domain name is `.wormhole.cluster.kyma.cx`.
-- **certificateServiceHost** - Host at which this service is accessible, used for URL generating. Default host is `cert-service.wormhole.cluster.kyma.cx`.
+- **appTokenExpirationMinutes** - Time after which tokens for applications expire and are no longer valid. The default value is `5` minutes.
+- **runtimeTokenExpirationMinutes** - Time after which tokens for runtimes expire and are no longer valid. The default value is `10` minutes.
+- **caSecretName** - Name of the secret which contains root CA. The default value if `nginx-auth-ca`
```
[the remark this answers is not in the document]
Comment (akgalwas) [reply components/connector-service/README.md]: Done

File: components/connector-service/internal/clientcontext/middlewares/applicationcontext_test.go [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+	})
+
+	t.Run("should set context based on header and cluster middleware", func(t *testing.T) {
+		// given
+		handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			ctx := r.Context()
+			clusterCtx, ok := ctx.Value(clientcontext.ApplicationContextKey).(clientcontext.ApplicationContext)
+			require.True(t, ok)
+
+			assert.Equal(t, testApplication, clusterCtx.Application)
```
[the remark this answers is not in the document]
Comment (akgalwas) [reply components/connector-service/internal/clientcontext/middlewares/applicationcontext_test.go]: Done

File: components/connector-service/internal/clientcontext/middlewares/applicationcontext_test.go [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+	emptyClusterContextMiddleware := &clusterContextMiddleware{}
+
+	t.Run("should set context based on header", func(t *testing.T) {
+		// given
+		handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			ctx := r.Context()
+			clusterCtx, ok := ctx.Value(clientcontext.ApplicationContextKey).(clientcontext.ApplicationContext)
+			require.True(t, ok)
+
+			assert.Equal(t, testApplication, clusterCtx.Application)
```
[the remark this answers is not in the document]
Comment (akgalwas) [reply components/connector-service/internal/clientcontext/middlewares/applicationcontext_test.go]: Done.

File: components/connector-service/internal/tokens/resolver_test.go [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+	"github.com/stretchr/testify/require"
+)
+
+func TestTokenService_Resolve(t *testing.T) {
+
+	type data struct {
+		Data string `json:"data"`
+	}
+
+	t.Run("shoud resolve token", func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (akgalwas) [reply components/connector-service/internal/tokens/resolver_test.go]: Done

File: components/connector-service/README.md [hunk 10]
```diff
@@ -25,9 +25,16 @@ The Connector Service has the following parameters:
 - **internalAPIPort** - This port exposes the Connector Service within Kubernetes cluster. The default port is `8080`.
 - **namespace** - Namespace where Connector Service is deployed. The default namespace is `kyma-integration`.
 - **tokenLength** - Length of registration tokens. The default value is `64`.
-- **tokenExpirationMinutes** - Time after which tokens expire and are no longer valid. The default value is `5` minutes.
-- **domainName** - Domain name of the cluster, used for URL generating. Default domain name is `.wormhole.cluster.kyma.cx`.
-- **certificateServiceHost** - Host at which this service is accessible, used for URL generating. Default host is `cert-service.wormhole.cluster.kyma.cx`.
+- **appTokenExpirationMinutes** - Time after which tokens for applications expire and are no longer valid. The default value is `5` minutes.
+- **runtimeTokenExpirationMinutes** - Time after which tokens for runtimes expire and are no longer valid. The default value is `10` minutes.
+- **caSecretName** - Name of the secret which contains root CA. The default value is `nginx-auth-ca`
```
Comment (kazydek) [inline components/connector-service/README.md]:
```suggestion
- **caSecretName** - Name of the secret which contains the root Certification Authority (CA). The default value is `nginx-auth-ca`.
```

File: components/connector-service/README.md [hunk 11]
```diff
@@ -25,9 +25,16 @@ The Connector Service has the following parameters:
...
+- **requestLogging** - Flag for logging incoming requests. `False` by default.
```
Comment (kazydek) [inline components/connector-service/README.md]:
```suggestion
- **requestLogging** - Flag for logging incoming requests. It is set to `False` by default.
```

File: components/connector-service/README.md [hunk 12]
```diff
@@ -25,9 +25,16 @@ The Connector Service has the following parameters:
...
+- **connectorServiceHost** - Host at which this service is accessible, used for URL generating. Default host is `cert-service.wormhole.cluster.kyma.cx`.
```
Comment (kazydek) [inline components/connector-service/README.md]:
```suggestion
- **connectorServiceHost** - Host under which this service is accessible. It is used for generating the URL. The default host is `cert-service.wormhole.cluster.kyma.cx`.
```

File: components/connector-service/README.md [hunk 13]
```diff
@@ -25,9 +25,16 @@ The Connector Service has the following parameters:
...
+- **appRegistryHost** - Host at which this Application Registry is accessible. Default value is empty string.
```
Comment (kazydek) [inline components/connector-service/README.md]:
```suggestion
- **appRegistryHost** - Host under which this Application Registry is accessible. The default value is an empty string.
```

File: components/connector-service/README.md [hunk 14]
```diff
@@ -25,9 +25,16 @@ The Connector Service has the following parameters:
...
+- **eventsHost** - Host at which this Event Service is accessible. Default value is empty string.
+- **getInfoURL** - URL at which management information is available. If not provided it will be constructed based on `connectorServiceHost`.
```
Comment (kazydek) [inline components/connector-service/README.md]:
```suggestion
- **getInfoURL** - URL at which the management information is available. If not provided, it bases on `connectorServiceHost`.
```

File: components/connector-service/README.md [hunk 15]
```diff
@@ -25,9 +25,16 @@ The Connector Service has the following parameters:
...
-- **domainName** - Domain name of the cluster, used for URL generating. Default domain name is `.wormhole.cluster.kyma.cx`.
-- **certificateServiceHost** - Host at which this service is accessible, used for URL generating. Default host is `cert-service.wormhole.cluster.kyma.cx`.
+- **appTokenExpirationMinutes** - Time after which tokens for applications expire and are no longer valid. The default value is `5` minutes.
+- **runtimeTokenExpirationMinutes** - Time after which tokens for runtimes expire and are no longer valid. The default value is `10` minutes.
+- **caSecretName** - Name of the secret which contains root CA. The default value is `nginx-auth-ca`
+- **requestLogging** - Flag for logging incoming requests. `False` by default.
+- **connectorServiceHost** - Host at which this service is accessible, used for URL generating. Default host is `cert-service.wormhole.cluster.kyma.cx`.
+- **appRegistryHost** - Host at which this Application Registry is accessible. Default value is empty string.
+- **eventsHost** - Host at which this Event Service is accessible. Default value is empty string.
```
Comment (kazydek) [inline components/connector-service/README.md]:
```suggestion
- **eventsHost** - Host under which this Event Service is accessible. The default value is an empty string.
```

File: docs/application-connector/docs/02-02-connector-service.md [hunk 16]
```diff
@@ -3,7 +3,7 @@ title: Connector Service
 type: Architecture
 ---
 
-The Connector Service generates client certificates which secure the communication between Kyma and the connected external solutions.        
+The Connector Service generates client certificates which secure the communication between Kyma and the connected external solutions as well as other Kyma clusters.
```
Comment (kazydek) [inline docs/application-connector/docs/02-02-connector-service.md]: As discussed - it seems unclear what is meant here.

File: components/connector-service/README.md [hunk 17]
```diff
@@ -25,9 +25,16 @@ The Connector Service has the following parameters:
...
+- **getInfoURL** - URL at which management information is available. If not provided it will be constructed based on `connectorServiceHost`.
+- **group** - Group for which certificates will be generated. If not provided, it will be required in the request header to token endpoint.
```
Comment (kazydek) [inline components/connector-service/README.md]:
```suggestion
- **group** - Group for which certificates are generated. If the chart does not provide the default value, you must specify it in the request header to the token endpoint.
```

File: components/connector-service/README.md [hunk 18]
```diff
@@ -25,9 +25,16 @@ The Connector Service has the following parameters:
...
+- **tenant** - Tenant for which certificates will be generated. If not provided, it will be required in the request header to token endpoint.
```
Comment (kazydek) [inline components/connector-service/README.md]:
```suggestion
- **tenant** - Tenant for which certificates are generated. If the chart does not provide the default value, you must specify it in the request header to the token endpoint.
```

**Pull request merged**